### PR TITLE
Mark `make_ascii_uppercase` and `make_ascii_lowercase` in `[u8]` and `str` as const.

### PR DESCRIPTION
**Pull request by bjoernager**
Relevant tracking issue: #130698

This PR extends #130697 and #130713 to the similar methods in byte slices (`[u8]`) and string slices (`str`). 

For the `str` methods, this simply requires adding the `const` specifier to the function signatures. The `[u8]` methods, however, require (at least a temporary) reimplementation due to the use of iterators and `for` loops.

### CODE REVIEW

**Comment by bjoernager** [comment]
@rustbot label +T-libs-api -T-libs

r? libs-api

**Comment by bjoernager** [comment]
No, what, how did I close this?!

**Comment by bjoernager** [comment]
This should not have been closed.

File: library/core/src/slice/ascii.rs [hunk 1]
```diff
@@ -67,10 +67,15 @@ impl [u8] {
     ///
     /// [`to_ascii_uppercase`]: #method.to_ascii_uppercase
     #[stable(feature = "ascii_methods_on_intrinsics", since = "1.23.0")]
+    #[rustc_const_unstable(feature = "const_make_ascii", issue = "130698")]
     #[inline]
-    pub fn make_ascii_uppercase(&mut self) {
-        for byte in self {
+    pub const fn make_ascii_uppercase(&mut self) {
+        // FIXME(const-hack): We would like to simply iterate using `for` loops but this isn't currently allowed in constant expressions.
+        let mut i = 0x0;
```
Comment (jhpratt) [inline library/core/src/slice/ascii.rs:74]: Why hex literals?

Comment (bjoernager) [reply library/core/src/slice/ascii.rs:74]: Sorry, I overlooked those. It's a habit.

**Pull request merged**